src/envs/painting.py

```diff
@@ -490,15 +497,25 @@ def _get_tasks(self,
                     pose[0], pose[1], pose[2], dirtiness, wetness, color, held
                 ],
                                      dtype=np.float32)
-                # last object should go in box
+                # Last object should go in box
                 if use_box and j == num_objs - 1:
                     goal.add(GroundAtom(self._InBox, [obj, self._box]))
                     goal.add(GroundAtom(self._IsBoxColor, [obj, self._box]))
                 else:
                     goal.add(GroundAtom(self._InShelf, [obj, self._shelf]))
                     goal.add(GroundAtom(self._IsShelfColor,
                                         [obj, self._shelf]))
-            tasks.append(Task(State(data), goal))
+            state = State(data)
+            # Sometimes start out holding an object, possibly with the wrong
+            # grip, so that we'll have to put it on the table and regrasp
+            if rng.uniform() < 0.5:
```

Comment:
same as in the Cover PR #327 , we should make this a setting and add a test specifically for it